packages/analytics-js/src/components/userSessionManager/UserSessionManager.ts

```diff
@@ -33,36 +122,47 @@ class UserSessionManager implements IUserSessionManager {
    *    using parseLinker util.
    * 3. generateUUID: A new unique id is generated and assigned.
    */
-  setAnonymousId(anonymousId?: string, rudderAmpLinkerParam?: string): string {
-    const anonymousIdFromLinker = defaultPluginManager.invoke<Nullable<string>>(
-      'userSession.anonymousIdGoogleLinker',
-      rudderAmpLinkerParam,
-    );
+  setAnonymousId(anonymousId?: string, rudderAmpLinkerParam?: string) {
+    let finalAnonymousId: string | undefined = anonymousId;
+    if (!finalAnonymousId) {
+      const linkerPluginsResult = defaultPluginManager.invoke<Nullable<string>>(
+        'userSession.anonymousIdGoogleLinker',
+        rudderAmpLinkerParam,
+      );
+      finalAnonymousId = linkerPluginsResult?.[0] || generateUUID();
+    }
 
-    state.session.rl_anonymous_id.value = anonymousId || anonymousIdFromLinker[0] || generateUUID();
-    this.storage?.set('rl_anonymous_id', state.session.rl_anonymous_id.value);
-    return state.session.rl_anonymous_id.value;
+    state.session.rl_anonymous_id.value = finalAnonymousId;
   }
 
+  /**
+   * Fetches anonymousId
+   * @param options option to fetch it from external source
+   * @returns anonymousId
+   */
   getAnonymousId(options?: AnonymousIdOptions): string {
     // fetch the rl_anonymous_id from storage
-    let persistedAnonymousId = this.storage?.get('rl_anonymous_id');
+    let persistedAnonymousId = this.storage?.get(persistedSessionStorageKeys.anonymousUserId);
 
     if (!persistedAnonymousId) {
-      // TODO: implement the storage.getAnonymousId autoCapture functionality as plugin that takes options in
-      const autoCapturedAnonymousId = [undefined];
-
-      if (autoCapturedAnonymousId[0]) {
-        persistedAnonymousId = autoCapturedAnonymousId[0];
+      // fetch anonymousId from external source
+      const autoCapturedAnonymousId = defaultPluginManager.invoke<string | undefined>(
+        'storage.getAnonymousId',
+        options,
+      );
+      if (autoCapturedAnonymousId?.[0]) {
+        [persistedAnonymousId] = autoCapturedAnonymousId;
       } else {
-        return this.setAnonymousId();
+        // set a new anonymousId if not available from previous step
+        this.setAnonymousId();
```

Comment:
Let's not use this API. Add a new function, `generateAnonymousId` and return the ID.

Reply:
Done

Reply:
Addressed